integration-tests/src/lib.rs

```diff
@@ -140,55 +188,153 @@ where
 pub fn initialize_test_logging_fallback() {
     static ONCE: Once = Once::new();
     ONCE.call_once(|| {
-        let var = env::var_os("XAYN_TEST_FALLBACK_LOG");
-        let directives = var
-            .as_deref()
-            .map(|s| {
-                s.to_str()
-                    .expect("XAYN_TEST_FALLBACK_LOG must only contain utf-8")
-            })
-            .unwrap_or("warn");
-        tracing_subscriber::fmt()
-            .with_ansi(false)
-            .with_writer(TestWriter::default())
-            .with_env_filter(directives)
-            .init()
+        let env_filter = env_opt_var_os(XAYN_TEST_FALLBACK_LOG)
+            .unwrap_or_else(|| "warn".into())
+            .parse::<EnvFilter>()
+            .expect("XAYN_TEST_FALLBACK_LOG has invalid EnvFilter directives");
+
+        let max_level = env_filter.max_level_hint();
+
+        // WARNING: You can not use `.init()` here as it will setup the log=>tracing
+        //          facade in a way which yields incorrect results.
```

Comment:
`.init()` setup the the `log` => `tracing` facade with the max level of the global logger, but we need to use the max level of _any_ global or non global  filter or log entries will be missing, especially given that we default to global fallback logger to `"warn"`